plugins/bufferlines/barbecue.nix

```diff
@@ -1,34 +1,72 @@
 {
   lib,
-  helpers,
-  config,
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.barbecue;
-  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
-in
-{
-  options.plugins.barbecue = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "barbecue-nvim";
+  inherit (lib.nixvim) defaultNullOpts;
 
-    package = helpers.mkPluginPackageOption "barbecue-nvim" pkgs.vimPlugins.barbecue-nvim;
-
-    attachNavic = helpers.defaultNullOpts.mkBool true ''
+  mkListStr = defaultNullOpts.mkNullable (lib.types.listOf lib.types.str);
```

Comment:
1. Is this abstraction really needed?
2. This doesn't support rawLua

I think we can just use `mkListOf types.str` throughout?